src/compiler/transformers/ts.ts

```diff
@@ -1943,13 +1943,19 @@ namespace ts {
             }
 
             let statements: Statement[] = [];
-            let indexOfFirstStatement = 0;
 
             resumeLexicalEnvironment();
 
-            indexOfFirstStatement = addPrologueDirectivesAndInitialSuperCall(factory, constructor, statements, visitor);
+            const { foundSuperStatement, indexOfFirstStatementAfterSuper, indexAfterLastPrologueStatement } = addPrologueDirectivesAndInitialSuperCall(factory, constructor, statements, visitor);

```

Comment:
As I stated below for `addPrologueDirectivesAndInitialSuperCall`, I think it would be better if we just broke this down into the component steps and inlined them here:

1. Copy/transform standard prologue directives to the new statement body using `factory.copyPrologue`
2. Find the index of the first `super` statement, and copy/transform statements between the prologue and the `super` call.
3. Transform the `super` call
4. Transform initializers and parameter-property assignments.
5. Transform the remaining statements.

That way we don't have to jump around within `statements` and prefix some statements and postifx others like we're doing on lines 1951-1956 and 1973-1980 as its just becoming to difficult to reason over.